includes/class-wc-geolocation.php

```diff
@@ -207,7 +207,7 @@ public static function geolocate_ip( $ip_address = '', $fallback = false, $api_f
 				$country_code = strtoupper( sanitize_text_field( wp_unslash( $_SERVER['HTTP_X_COUNTRY_CODE'] ) ) ); // WPCS: input var ok, CSRF ok.
 			} else {
 				$ip_address = $ip_address ? $ip_address : self::get_ip_address();
-				$database   = self::get_local_database_path();
+				$database   = WC_MaxMind_Geolocation_Database::get_database_path();
```

Comment:
This also could be changed by a filter, so we can filter by the integration.
Also we can remove `supports_geolite2` and `geolocate_via_db`.

Reply:
The idea is to make this class let's directly dependent on an integration.

Reply:
Also missing put a deprecate notice about `GEOLITE2_DB`.

Reply:
Good catch!